core/templates/components/state-directives/outcome-editor/outcome-destination-editor.directive.ts

```diff
@@ -73,28 +73,12 @@ angular.module('oppia').directive('outcomeDestinationEditor', [
             ctrl.maxLen = MAX_STATE_NAME_LENGTH;
             return outcome.dest === PLACEHOLDER_OUTCOME_DEST;
           };
-          ctrl.$onInit = function() {
-            ctrl.directiveSubscriptions.add(
-              StateEditorService.onSaveOutcomeDestDetails.subscribe(
-                () => {
-                  if (ctrl.isSelfLoop()) {
-                    ctrl.outcome.dest = StateEditorService.getActiveStateName();
-                  }
-                  // Create new state if specified.
-                  if (ctrl.outcome.dest === PLACEHOLDER_OUTCOME_DEST) {
-                    EditorFirstTimeEventsService
-                      .registerFirstCreateSecondStateEvent();
-
-                    var newStateName = ctrl.outcome.newStateName;
-                    ctrl.outcome.dest = newStateName;
-                    delete ctrl.outcome.newStateName;
 
-                    ctrl.addState(newStateName);
-                  }
-                }
-              )
-            );
-            $scope.$watch(StateEditorService.getStateNames, function() {
```

Comment:
@srijanreddy98 found that using an rxjs Subscription rather than depending on $scope.$watch resolved the e2e test errors in this PR.